### ISSUE TITLE
use external disk

### ISSUE THREAD

**Issue by neulf** [nextcloud/nextcloud-snap#767]
I want to use external path. eg. /data1/nextcloud/data. I do it according to the follow. but not ok.

If you just installed the snap, and haven't created an admin user yet
Decide where you want the new data to live. We'll use /media/my/new/data. Update the Nextcloud config to use it by editing /var/snap/nextcloud/current/nextcloud/config/autoconfig.php and making sure the directory setting is pointing to the right place, e.g.:
 // ...
 'directory' => '/media/my/new/data',
 // ...
Now, restart the PHP service:
 $ sudo snap restart nextcloud.php-fpm
If you visited Nextcloud before now, refresh to re-evaluate the changed config (otherwise the directory won't change from the default).
Make your admin user as usual and carry on, Nextcloud is using the external data.

If you have already created an admin user
Decide where you want the new data to live. We'll use /media/my/new/data. Update the Nextcloud config to use it by editing /var/snap/nextcloud/current/nextcloud/config/config.php and making sure the datadirectory setting is pointing to the right place, e.g.:
 // ...
 'datadirectory' => '/media/my/new/data',
 // ...
Stop the snap from running for a moment:
 $ sudo snap disable nextcloud
Move (or copy) the current data directory to the new place:
 $ sudo mv /var/snap/nextcloud/common/nextcloud/data /media/my/new/data
Re-enable the snap:
 $ sudo snap enable nextcloud

**Comment by JamesBream** [comment]
So... What's your problem?

**Comment by neulf** [comment]
> Can you give a bit more detail around what your situation is? Did you see any errors for example?

Internal server error after moving Nextcloud snap data directory

**Comment by kyrofa** [comment]
Did you connect the `removable-media` interface?

**Comment by arjunak234** [comment]
I too have the same problem
```
# grep 'datadirectory' /var/snap/nextcloud/current/nextcloud/config/config.php 
  #'datadirectory' => '/var/snap/nextcloud/common/nextcloud/data/',
  'datadirectory' => '/hdd/nextcloud/data/',

# ll /hdd/nextcloud/data/.ocdata 
-rw-r--r-- 1 root root 0 Nov 10 23:29 /hdd/nextcloud/data/.ocdata

# snap interfaces nextcloud
Slot              Plug
:network          nextcloud
:network-bind     nextcloud
:removable-media  nextcloud
```

**Comment by Chrozayis** [comment]
Trying to set the data directory with
`sudo nextcloud.occ config:system:set datadirectory /media/nextcloud/data`
results in
`Your data directory must be an absolute path
Check the value of "datadirectory" in your configuration

Your data directory is invalid
Ensure there is a file called ".ocdata" in the root of the data directory.

Cannot create "data" directory
This can usually be fixed by giving the webserver write access to the root directory. See https://docs.nextcloud.com/server/13/go.php?to=admin-dir_permissions

An unhandled exception has been thrown:
Exception: Environment not properly prepared. in /snap/nextcloud/10314/htdocs/lib/private/Console/Application.php:148
Stack trace:
#0 /snap/nextcloud/10314/htdocs/console.php(89): OC\Console\Application->loadCommands(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#1 /snap/nextcloud/10314/htdocs/occ(11): require_once('/snap/nextcloud...')`
Attempting to set the directory back to it's original location also results in this error.

/media/nextcloud/data does contain the .ocdata file and the user data files.
The link in the error message gets redirected and is unhelpful.

My install of Nextcloud is currently bricked because of this.

**Comment by kyrofa** [comment]
Any chance those errors are actually old, from when the setting was incorrect?

**Comment by Chrozayis** [comment]
Almost certainly, I've determined where my issues came from.
It started when I thought the config file didn't exist in the expected location when I tried using vim rather than sudo vim to access it.
I then tried to use nextcloud.occ to set the value, I believe an error in nextcloud.occ compounded my problems. I will do further testing and post a separate report.
After discovering that I needed root privileges to edit config.php I was able to resolve all except some warnings on the admin panel.
